checkerScripts.js

```diff
@@ -134,40 +139,96 @@ function getPieceMovement(){
+	var topRightBoxIndex = (selectedPieceColor == 1) ? ( ( (crdX > 0) && (crdY+1 < 8) ) ? (crdX-1) + '' + (crdY+1) : null)
+                                                     : ( ( (crdX+1 < 8) && (crdY+1 < 8) ) ? (crdX+1) + '' + (crdY+1) : null);
+
 	selectedPiece = crdX + '' + crdY;
-	coloredPiecePlaces(topLeftBoxIndex, topRightBoxIndex);
+    console.log(topLeftBoxIndex, topRightBoxIndex);
+	
+    coloredPiecePlaces(topLeftBoxIndex, topRightBoxIndex);
 }
 
 function coloredPiecePlaces(left, right) {
-	if(left != null)
-		if( board[left[0]][left[1]] == 0){
-			$(this).css("background-color", "yellow");
-			$('.white-box[name=' + left + ']').css("background-color", "yellow");
-			leftPiecePlace = left;
-		}
-
 
-	if(right != null)
-		if(board[right[0]][right[1]] == 0){
+	if(left != null){
+        var leftX = parseInt(left[0]);
+        var leftY = parseInt(left[1]);
+
+		if( (leftX >= 0 && leftY >= 0) && board[leftX][leftY] == 0 && right != null && board[right[0]][right[1]] == 0){
+    		$(this).css("background-color", "yellow");
+    		$('.white-box[name=' + left + ']').css("background-color", "yellow");
+    		leftPiecePlace = left;
+            
+		}else if(leftX >=0 && leftY >=0 && board[leftX][leftY] != 0)
+            coloredOnJumpPiece(leftX, leftY, 'L');
+    }
+    
+
+	if(right != null){
+        var rightX = parseInt(right[0]);
+        var rightY = parseInt(right[1]);
+
+		if( (rightX < 8 && rightY < 8) && board[rightX][rightY] == 0 && left != null && board[left[0]][left[1]] == 0){
 			$(this).css("background-color", "yellow");
 			$('.white-box[name=' + right + ']').css("background-color", "yellow");
-			rigthPiecePlace = right;		
-	}
+			rigthPiecePlace = right;
+	    
+        }else if(rightX < 8 && rightY < 8 && board[rightX][rightY] != 0)
+            coloredOnJumpPiece(rightX, rightY, 'R');
+    }
+
 
 }
 
 
+function coloredOnJumpPiece(x, y, side){
+    var place = null;
+    
+    if(selectedPieceColor == 1 && side == 'R')
+        place = (board[x-1][y+1] == 0 && board[x][y] == -1) ? (x-1).toString() + (y+1).toString() : null;
+    
+    else if(selectedPieceColor == 1 && side == 'L')
+        place = (board[x-1][y-1] == 0 && board[x][y] == -1) ? (x-1).toString() + (y-1).toString() : null;
+    
+    else if(selectedPieceColor == -1 && side == 'R')
+        place = (board[x+1][y+1] == 0  && board[x][y] == 1) ? (x+1).toString() + (y+1).toString() : null;
+    
+    else if(selectedPieceColor == -1 && side == 'L')
+        place = (board[x+1][y-1] == 0 && board[x][y] == 1) ? (x+1).toString() + (y-1).toString() : null;
+    
+    
+
+    $('.white-box[name=' + place + ']').css("background-color", "yellow");
+
+    rigthPiecePlace = (side == 'R') ? place : rigthPiecePlace;
+    leftPiecePlace = (side == 'L') ? place : leftPiecePlace;
+
+    if(side == 'L')
+        pieceRemove = (x).toString()+(y).toString();
+    else
+        pieceRemove = (x).toString()+(y).toString();
```

Comment:
Please check this `if` condition, you are assigning the same value to `pieceRemove` variable in both cases.